src/main/java/pl/szczesniak/dominik/whattowatch/movies/infrastructure/persistence/InMemoryMoviesRepository.java

```diff
@@ -0,0 +1,51 @@
+package pl.szczesniak.dominik.whattowatch.movies.infrastructure.persistence;
+
+import lombok.ToString;
+import pl.szczesniak.dominik.whattowatch.movies.domain.Movie;
+import pl.szczesniak.dominik.whattowatch.movies.domain.MoviesRepository;
+import pl.szczesniak.dominik.whattowatch.movies.domain.model.MovieId;
+import pl.szczesniak.dominik.whattowatch.users.domain.model.UserId;
+
+import java.util.HashMap;
+import java.util.List;
+import java.util.Map;
+import java.util.concurrent.atomic.AtomicInteger;
+import java.util.stream.Collectors;
+
+@ToString
+public class InMemoryMoviesRepository implements MoviesRepository {
+
+	private final AtomicInteger nextId = new AtomicInteger(0);
+	private final Map<MovieId, Movie> movies = new HashMap<>();
+
+	@Override
+	public MovieId nextMovieId() {
+		return new MovieId(nextId.incrementAndGet());
+	}
+
+	@Override
+	public void save(final Movie movie) {
+		movies.put(movie.getMovieId(), movie);
+	}
+
+	@Override
+	public List<Movie> findAll(final UserId userId) {
+		return movies.values().stream()
+				.filter(movie -> movie.getUserId().equals(userId))
+				.collect(Collectors.toList());
+	}
+
+	@Override
+	public void removeMovie(final MovieId movieId, final UserId userId) {
+		if (movieIdDoesntMatchUserId(movieId, userId)) {
+			System.out.println("Didn't remove movie.");
+		} else {
+			movies.remove(movieId);
+		}
+	}
+
+	private boolean movieIdDoesntMatchUserId(final MovieId movieId, final UserId userId) {
```

Comment:
zazwyczaj lepiej robić metody bez zaprzeczeń movieIdDoesntMatchUserId -> movieMatchesUserId, albo nawet lepiej movieBelongsToUser

i możesz tego ifa odwrócić - jeśli film jest usera to usuwamy, a jak nie to nie